src/main/java/gnu/io/RXTXPort.java

```diff
@@ -871,94 +904,54 @@ public void addEventListener(
 			throw new TooManyListenersException();
 		}
 		SPEventListener = lsnr;
-		if( !MonitorThreadAlive )
+		if (this.monThread == null)
 		{
-			MonitorThreadLock = true;
+			this.monitorThreadState.writeLock().lock();
 			monThread = new MonitorThread();
 			monThread.setName("RXTXPortMonitor("+name+")");
 			monThread.start();
-			waitForTheNativeCodeSilly();
-			MonitorThreadAlive=true;
+			try {
+				this.monitorThreadReady.await();
+			} catch (InterruptedException e) {
+				z.reportln("Interrupted while waiting for the monitor thread to start!");
+			}
+			this.monitorThreadState.writeLock().unlock();
```

Comment:
This might be a good place to check whether the monitoring thread's initialization phase was successful.  
(If it wasn't, throw an exception and clean up the state so the client has a chance to retry.)